client/src/components/Tool/utilities.js

```diff
@@ -2,7 +2,9 @@ import { getAppRoot } from "onload/loadConfig";
 import { copy } from "utils/clipboard";
 
 export function copyLink(toolId, message) {
-    copy(`${window.location.origin + getAppRoot()}root?tool_id=${toolId}`, message);
+    const link = `${window.location.origin + getAppRoot()}root?tool_id=${toolId}`;
+    // Encode the link to handle special characters in tool id
+    copy(encodeURI(link), message);
```

Comment:
Are you sure that doesn't break tools with `+` in the tool id ? Can we maybe just replace space with percent encoding ?

Reply:
Good point. I just checked and `encodeURI` should be fine with `+` according to the docs in https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURI#encodeuri_vs._encodeuricomponent

I will add another test just to make 100% sure.

Any other possible special character that can be contained in a tool ID?